hardware_interface/include/hardware_interface/control_type.hpp

```diff
@@ -0,0 +1,30 @@
+// Copyright 2020 ros2_control development team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HARDWARE_INTERFACE__CONTROL_TYPE_HPP_
+#define HARDWARE_INTERFACE__CONTROL_TYPE_HPP_
+
+namespace hardware_interface
+{
+
+namespace joint_control_type
+{
+constexpr const auto POSITION = "PositionJointInterface";
+constexpr const auto VELOCITY = "VelocityJointInterface";
+constexpr const auto EFFORT = "EffortJointInterface";
```

Comment:
Following the latest discussions from the WG, I think to believe we agreed of not hardcoding this triplet. I also thought that we've agreed that, if hardcoded, we declare them as constants in the flexible joint state message as "default" joint interfaces.

Reply:
+1

Reply:
I think you guys misunderstood this one. Yes, the original PR also had these values hardcoded but the method and how entrenched that was is very different: Encoded into an enum and used in the transmission info, doing the string-to-enum conversion in the parsing function is.

This on the other hand hardcoded the *standard names* we want to maintain which I'd argue is quite relevant. We can discuss moving these at some point to the message but I'd prefer to avoid breaking changes in `control_msgs`.

The interface names are returned as a string by the parser and these constants will be used to instantiate them down the line.

Reply:
well, at some point we have to "break" API for the control_msgs if we want to have these constants. Rather do it now before the Foxy release than while Foxy being out.

Reply:
I'm not comfortable starting a multi-repo, from-source collaboration of half-baked things. Here we can curate what set we are happy with, then add to `control_msgs` with no harm.
It's unreasonable to ask to smudge this information around at this point.

Reply:
https://github.com/ros-controls/ros2_control/issues/95

Reply:
@bmagyar what do you think about creating development branch in control_msgs so we can work on it. In that way we should avoid any "include" changes in the later process.

Reply:
You just asked the same thing as above. I'd still like to avoid inter repo dependencies and then building everything from source for something this simple.